include/internal/crypto.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#ifndef HEADER_INTERNAL_CRYPTO_H
+# define HEADER_INTERNAL_CRYPTO_H
+
+# if __STDC_VERSION__ >= 201112L && !defined(__STDC_NO_ATOMICS__)
+
+# include <stdatomic.h>
+
+#define HAVE_ATOMICS 1
+
+typedef _Atomic int crypto_ref_count;
```

Comment:
I don't think you need `_Atomic` when using the `atomic_` operations, unless you want to support an hybrid thing where people can do e.g. `obj->refcount++`, but in that case you wouldn't need the `atomic_` part instead (also, `atomic_int` is a thing IIRC).